services/scheduler/scheduler.go

```diff
@@ -81,9 +85,12 @@ type AddParams struct {
 // Add adds task to scheduler and save it to DB.
 func (s *Service) Add(task Task, params AddParams) (*models.ScheduledTask, error) {
 	var scheduledTask *models.ScheduledTask
-	var err error
+	errTx := s.db.InTransactionContext(s.db.Querier.Context(), &sql.TxOptions{Isolation: sql.LevelSerializable}, func(tx *reform.TX) error {
```

Comment:
Important stuff! This transaction is correct only on a serializable level. We need it because of `checkPreconditions` function, where we are checking if there other enabled scheduled backups for that service.

So simplified tx is:
```
// Chech existing tasks
SELECT * FROM scheduled_tasks WHERE ; 
// If the result is empty then we can enable a new task;
INSERT INTO tasks …;
```

On other isolation levels it's possible to get two enable PITR tasks.

Reply:
so, let's add it as comment

Reply:
Done